errors.go

```diff
@@ -4,7 +4,29 @@ import (
 	"fmt"
 )
 
+type RecoverableError interface {
+	error
+	CanRecover() bool
```

Comment:
I'm not wild about this approach, since it requires plumbing a different type in a bunch of places, and whether an error is 'recoverable' or not seems to depend a lot on context.

I think it's be cleaner to just have in `transport.go`:

```
func shouldContinue(err error) {
  if err == nil {
    return true
  }
  // We may get these when we cancel a call after the server has already sent its reply.
  if _, ok := err.(CallNotFoundError); ok {
    return true
  }
  return false
}
```

and use that in the packetizer loop.